### PR TITLE
Ensure Persistent File Browsing Location

### PR DESCRIPTION
**Pull request by capricornunicorn123**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Refactors the `FileInputPane` to use the static `FileChooser.openDialogue()` instead of the `FileChooser` instantiation with method `showDialogue()`.

The property "Remember open save location" has been removed.

Plugins that used an `ExtensionFilter` have been reviewed to ensure that they utilize the `FileExtensionConstants` class.

`FileParameterKind`s have been updated to include an "obscured" alternate to existing file kinds. This enables the same functionality as their non obscured alternative but enables the button to express the text "..." instead of "save" or "open".

### Why Should This Be In Core?

It related to core functionality

### Benefits

Ensure that the modified file exporters use the same methods / logic.

### Verification Process

Open, Save, Import and Export files within the application and note the persistence of the directory upon executing the plugin. 

### Applicable Issues

#1825 
#1721

### CODE REVIEW

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java [hunk 1]
```diff
@@ -83,41 +89,59 @@ public FileInputPane(final PluginParameter<FileParameterValue> parameter, final
...
-                        files.add(saveFile);
-                    }
+                case SAVE_OBSCURED:
+                    dialogFuture = FileChooser.openSaveDialog(getFileChooser(parameter, "Save")).thenAccept(optionalFile -> optionalFile.ifPresent(saveFile -> {
+                        if (saveFile != null) {
+                            
+                            //Save files may have been typed by the user and an extension may not have been specified.
+                            final String fnam = saveFile.getAbsolutePath();
+
+                            String expectedExtension = FileParameterType.getFileFilters(parameter).getExtensions().get(0);
```
Comment (Delphinus8821) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java:121]: final

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java [hunk 2]
```diff
@@ -83,41 +89,59 @@ public FileInputPane(final PluginParameter<FileParameterValue> parameter, final
...
             }
-
-            if (!files.isEmpty()) {
+            
+            // As the dialog windows are completed on another thread 
+            // the execution of this method must wait until the thread has finnished executing.
+            if (dialogFuture != null){
+                try {
+                    dialogFuture.get();
+                } catch (InterruptedException | ExecutionException ex) {
```
Comment (Delphinus8821) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java:140]: final and should split this into two and interrupt the thread on the interrupted exception

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java [hunk 3]
```diff
@@ -83,44 +89,64 @@ public FileInputPane(final PluginParameter<FileParameterValue> parameter, final
...
             }
-
-            if (!files.isEmpty()) {
+            
+            // As the dialog windows are completed on another thread 
+            // the execution of this method must wait until the thread has finnished executing.
+            if (dialogFuture != null){
+                try {
+                    dialogFuture.get();
+                } catch (InterruptedException ex){
```
Comment (antares1470) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java:140]: exceptions can be final

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/file/FileExtensionConstants.java [hunk 4]
```diff
@@ -57,15 +58,24 @@ private FileExtensionConstants() {
...
     public static final String CASCADING_STYLE_SHEET = ".css";
     public static final String BINARY = ".bin";
     public static final String BACKUP = ".bak";
-
+    
+    //Geographical file extensiona
+    public static final String GEO_PACKAGE = ".gpkg";
+    public static final String KML = ".kml";
+    public static final String SHAPE = ".shp";
+    public static final String GML = ".gml";
```
Comment (antares1470) [inline CoreUtilities/src/au/gov/asd/tac/constellation/utilities/file/FileExtensionConstants.java:73]: GML, GRAPHML, and PAJEK should be grouped together under Graph file extensions

**Review by antares1470 — CHANGES_REQUESTED** [review]
I gave it a test with the Import Graph File plugin. I'm not sure how I feel about the text "Open Multiple" as that seems 'straight out of the code' if you get what I mean. Maybe something like "Open File(s)" might be a little friendlier

**Review by Delphinus8821 — APPROVED** [review]

**Comment by capricornunicorn123** [comment]
> Maybe something like "Open File(s)" might be a little friendlier

Fair call, I've amended the text to "Open File(s)"

**Review by antares1470 — APPROVED** [review]

**Pull request merged**